### PR TITLE
Fix different issues in datatypes notebook

### LINKED ISSUE #225: Problems in the datatypes notebook
1. The correct solution for the following exercise doesn't work:

```python
%%ipytest

def solution_lists_are_equal(list1: list, list2: list) -> bool:
    # Your code starts here
    return list1 == list2
    # Your code ends here
```
It fails one test case.

2. This example should fail, but it doesn't in Python 3.11. We should add a note
```python
# This should fail.
2.5 + "1.5"
```

3. Exercises on sets are not checked? When I run the cells, nothing shows up.

### PR DESCRIPTION
**Pull request by yakutovicha**
fixes #225

### CODE REVIEW

**Comment by despadam** [comment]
I get the following error for these exercises:
1. `solution_addition_multiplication()`
2. `solution_circle_area()`
3. `solution_quadratic_equation()`

image

**Comment by despadam** [comment]
I get the following error for these exercises:
1. `solution_dict_return_value()`
2. `solution_dict_return_value_delete()`

image

**Review by despadam — CHANGES_REQUESTED** [review]
I noticed that some exercises are not running for me. Am I doing something wrong? Please, see my comments above.

**Comment by yakutovicha** [comment]
@despadam, I addressed the reported issues in https://github.com/empa-scientific-it/python-tutorial/pull/230/commits/5409bac44467d41929c6d90bbbe49a7ecf83cb17, please have a look

File: basic_datatypes.ipynb [hunk 1]
```diff
@@ -2085,9 +2083,9 @@
    "outputs": [],
    "source": [
     "%%ipytest\n",
-    "import typing\n",
-    "T = typing.Typevar(\"T\")\n",
-    "def solution_dict_return_value(my_dict: dict[typing.Hashable, T], key: typing.Hashable) -> typing.Optional[T]:\n",
+    "from typing import TypeVar, Hashable, Any\n",
+    "\n",
+    "def solution_dict_return_value(my_dict: dict[Hashable, Any], key: Hashable) -> Any:\n",
     "    # Your code starts here\n",
```
Comment (baffelli) [inline basic_datatypes.ipynb:2089]: Why did you remove the typevar? Now the type is "too generic"?

**Review by baffelli — APPROVED** [review]
It looks ok, but I don't see why we should remove the type variables.

File: basic_datatypes.ipynb (hunk 1 above)
Comment (yakutovicha) [reply basic_datatypes.ipynb:2089]: I thought the value of a dictionary could be any Python object, which I've done by setting the `Any` type. I do not understand the exact problem here.

The original implementation had a bug that @despadam reported above, so I looked around and this is what I found as the solution.

Comment (baffelli) [reply basic_datatypes.ipynb:2089]: It is correct, but if you use a type variable, then the function has a more specific return type. It does not really matter for the moment, so I suggest to merge anyway

**Review by despadam — APPROVED** [review]

File: basic_datatypes.ipynb (hunk 1 above)
Comment (despadam) [reply basic_datatypes.ipynb:2089]: It also looks ok to me and solves the bugs, so I think that since these exercises are anyway trivial, there is no need to worry about typing now
